grammar.js

```diff
@@ -477,92 +475,68 @@ grammar({
     // Expressions
 
     _expression: $ => choice(
-      $._statement,
       $._definition,
+      $._statement,
+      $._literal,
+      $._primary_expression,
+      $.operator,
       $.typed_expression,
       $.pair_expression,
       alias(':', $.operator),
-      $.macro_expression,
+      $.macrocall_expression,
       $.unary_expression,
       $.binary_expression,
       $.ternary_expression,
-      $.generator_expression,
       $.function_expression,
       $.coefficient_expression,
       $.spread_expression,
       $.range_expression,
-      $.quote_expression,
-      $.interpolation_expression,
-      $._primary_expression,
-      $._literal,
-      $.operator,
     ),
 
+    // Primary expressions can be called, indexed, accessed, and type parameterized.
     _primary_expression: $ => choice(
+      $._quotable,
+      $.broadcast_call_expression,
+      $.call_expression,
+      $.curly_expression,
+      $.field_expression,
+      $.index_expression,
+      $.interpolation_expression,
+      $.quote_expression,
+    ),
+
+    // Quotables are primary expressions that can be quoted without additional
+    // parentheses.
+    _quotable: $ => choice(
       $.identifier,
-      $.array_expression,
       $.array_comprehension_expression,
+      $.array_expression,
+      $.generator_expression,
       $.matrix_expression,
-      $.call_expression,
-      $.field_expression,
       $.parenthesized_expression,
-      $.subscript_expression,
-      $.parameterized_identifier,
       $.tuple_expression,
-      $.broadcast_call_expression,
-    ),
-
-    bare_tuple_expression: $ => prec(-1, seq(
-      $._expression,
-      repeat1(prec(-1, seq(',', $._expression)))
-    )),
-
-    operator: $ => choice(
-      $._comparison_operator,
-      $._dotty_operator,
-      $._plus_operator,
-      $._times_operator,
-      $._rational_operator,
-      $._bitshift_operator,
-      $._power_operator,
-      $._unary_operator,
     ),
 
-    parenthesized_expression: $ => prec(1, seq(
-      '(', choice($._block, $.spread_expression), ')'
-    )),
-
     field_expression: $ => prec(PREC.dot, seq(
       $._primary_expression,
       token.immediate('.'),
       $.identifier
     )),
 
-    subscript_expression: $ => seq(
-      choice(
-        $._primary_expression,
-        $._literal,
-      ),
+    index_expression: $ => seq(
+      $._primary_expression,
       token.immediate('['),
       sep(',', $._expression),
       optional(','),
       ']'
     ),
 
-    typed_expression: $ => prec(PREC.decl, seq(
-      $._expression,
-      choice('::', '<:'),
-      choice($.identifier, $.parameterized_identifier)
-    )),
-
-    parameterized_identifier: $ => seq(
-      choice($.identifier, $.field_expression),
-      $.type_argument_list
-    ),
 
-    type_argument_list: $ => seq(
+    curly_expression: $ => seq(
+      $._primary_expression,
       '{',
```

Comment:
```suggestion
      'token.immediate('{'),',
```

Reply:
> Do we really want to call it curly_expression? 

Yeah, `curly` isn't exactly the most useful name, but it's important it ends with `_expression` like the other primary expression. Maybe `parametrized_expression`? I prefer that spelling to `parameterized_` (but both are fine).

> I would also prefer to keep type_argument_list or something similar as a scope.

I think using field names for the object would be better for this, and we can add them to all primary expression. So the output would look like:

```scheme
; xs[i, j, …]
(index_expression 
  name: (identifier)
  (idenfier)
  (idenfier)
  …)

; T{R, S, …}
(parametric_expression 
  name: (identifier)
  (idenfier)
  (idenfier)
  …)

; etc.
```

Reply:
1. Think `paramerized_expression` might be a bit too general. Functions with `where` syntax would also fall under this. `parameterized_type_expression` is a bit wordy but should be fine. Could also think about dropping `_expression` altogether but this is probably better done later and in a separate discussion.

2. Using field names has the disadvantage that you lose the gradual selection/movement on the tree-sitter nodes. They are only useful for the queries. Most grammars seem to currently do it this way, so this is fine for me. We can also discuss this later in a different issue.

Reply:
I updated the name to `parametrized_type_expression`. I kept the "parametrized" spelling because it won't match `parameter*` when grepping.

I didn't add field names to call expressions because those already have an `argument_list`.

PS: I have to update assignments before adding the immediate brace in `parametrized_type_expression`, to avoid a conflict with `_function_signature`.